### DISCUSSION TITLE
Using SkiaSharp with AvaloniaUI Controls

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#13527] [Q&A] [answered]
Hello! I am interested in using AvaloniaUI since it has a Skia back end, and I am currently using Skia with WinForms, but really interested in Avalonia since it has XAML/MVVM support and can be AOT'd.

Is it possible to get a SkiaSharp surface/canvas context on a control in Avalonia? I've been googling/searching on this, and have been unable to find a clear/easy way of doing this. I noticed overriding Render gives me access to a lot of drawing primitives _similar_ to SkiaSharp, but was hoping for a bit more control.

I saw some older method that involve casting IDrawingContextImpl , but this doesn't seem possible anymore?

Is this possible?

Thank you!!

**Comment by jp2masa** [comment, marked as answer]
See this sample: https://github.com/AvaloniaUI/Avalonia/blob/a6e936d74accdcf7ead6b776f374defb501e4434/samples/RenderDemo/Pages/CustomSkiaPage.cs.

**Comment by ghost** [comment]
> See this sample: https://github.com/AvaloniaUI/Avalonia/blob/a6e936d74accdcf7ead6b776f374defb501e4434/samples/RenderDemo/Pages/CustomSkiaPage.cs.

Thanks!

**Comment by workgroupengineering** [comment]
and that https://github.com/AvaloniaUI/Avalonia.Labs/tree/main/src/Avalonia.Labs.Controls/SKCanvasView
